packages/stellar-axelar-std-derive/src/modifier.rs

```diff
@@ -0,0 +1,110 @@
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::ItemFn;
+
+pub fn modifier_impl(input_fn: ItemFn, auth_statement: TokenStream2) -> TokenStream2 {
+    let fn_vis = &input_fn.vis;
+    let fn_sig = &input_fn.sig;
+    let fn_name = &fn_sig.ident;
+    let fn_generics = &fn_sig.generics;
+    let fn_inputs = &fn_sig.inputs;
+    let fn_output = &fn_sig.output;
+    let fn_body = &input_fn.block;
+    let fn_attrs = &input_fn.attrs;
+
+    // Check that env is the first parameter
+    let Some(syn::FnArg::Typed(pat_type)) = fn_inputs.first() else {
+        panic!("First parameter must be a typed parameter")
+    };
+    let syn::Pat::Ident(pat_ident) = &*pat_type.pat else {
+        panic!("First parameter must be a simple identifier")
+    };
+    assert!(
+        pat_ident.ident == "env",
+        "First parameter must be named 'env'"
+    );
+
+    quote! {
+        #(#fn_attrs)*
+        #fn_vis fn #fn_name #fn_generics(#fn_inputs) #fn_output {
+            #auth_statement
+
+            #fn_body
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use syn::{parse_quote, ItemFn};
+
+    use super::*;
+
+    #[test]
+    fn test_first_parameter_is_typed() {
```

Comment:
Remove `test_` prefix
```suggestion
    fn first_parameter_is_typed() {
```